src/ucp/core/ucp_mm.c

```diff
@@ -583,13 +578,17 @@ ucs_status_t ucp_mem_map(ucp_context_h context, const ucp_mem_map_params_t *para
                               ucp_mem_map_params2uct_flags(params), memh_p);
     }
 
+    if (status == UCS_OK) {
+        (*memh_p)->map_count++;
+    }
+
 out:
     return status;
 }
 
 ucs_status_t ucp_mem_unmap(ucp_context_h context, ucp_mem_h memh)
 {
-    ucp_memh_put(context, memh, 1);
+    ucp_memh_put(context, memh, !--memh->map_count);
```

Comment:
i would expect that even when UCP performs the allocation, it would still go through system memory allocator (eventually), so it will also trigger memory hooks for release

Reply:
without rcache it would.
but rcache will keep this memory referenced by pgtable even after it was seemingly put and this will prevent it's release.
it just won't reach system dealocator

Reply:
i see, is it because we release from system memory only when rcache completely removes the region?
so, can we invalidate only in UCP rcache, but without calling uct_mem_dereg_v2(INVALIDATE) ?

Reply:
Sure, rcache invalidation has no direct relation to UCT invalidation.